pkg/machines/components/vm/memoryModal.jsx

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+import {
+    Alert,
+    Button,
+    Form,
+    FormGroup,
+    Grid,
+    Modal
+} from 'patternfly-react';
+import PropTypes from 'prop-types';
+
+import cockpit from 'cockpit';
+import * as Select from 'cockpit-components-select.jsx';
+import {
+    units,
+    convertToUnit,
+    digitFilter,
+    toFixedPrecision,
+    vmId
+} from '../../helpers.es6';
+import {
+    setMemory,
+    setMemorySettings,
+    getVm
+} from '../../actions/provider-actions.es6';
+
+import './memoryModal.css';
+
+const _ = cockpit.gettext;
+
+const clamp = (value, max, min) => {
+    return value < min ? min : (value > max ? max : value);
+};
+
+export class MemoryModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showModal: false,
+            dialogError: undefined,
+            /* What user might understand as 'memory' is defined in libvirt as 'current memory' */
+            memory: props.vm.currentMemory,
+            memoryUnit: units.MiB.name,
+            /* What user might understand as 'max memory' is defined in libvirt as 'memory' */
+            maxMemory: props.vm.memory,
+            maxMemoryUnit: units.MiB.name,
+        };
+        this.open = this.open.bind(this);
+        this.close = this.close.bind(this);
+        this.save = this.save.bind(this);
+        this.onMemoryBlur = this.onMemoryBlur.bind(this);
+        this.onMaxMemoryBlur = this.onMaxMemoryBlur.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.onUnitsValueChanged = this.onUnitsValueChanged.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.dialogErrorDismiss = this.dialogErrorDismiss.bind(this);
+    }
+
+    onMemoryBlur (memory) {
+        let maxMemory = convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'MiB');
+        memory = convertToUnit(memory, this.state.memoryUnit, 'MiB');
+
+        memory = clamp(memory, maxMemory, 1);
+
+        memory = convertToUnit(memory, 'MiB', this.state.memoryUnit);
+
+        this.setState({ memory: memory });
+    }
+
+    onMaxMemoryBlur (maxMemory) {
+        let memory = convertToUnit(this.state.memory, this.state.memoryUnit, 'MiB');
```

Comment:
Please take care of the blank lines here as well. I would add one after let and then the rest can be together without blank lines.